src/sentry/incidents/logic.py

```diff
@@ -716,21 +716,16 @@ def delete_alert_rule(alert_rule):
         AlertRuleStatus.DELETION_IN_PROGRESS.value,
     ):
         raise AlreadyDeletedError()
-    # TODO: We're assuming only one subscription for the moment
-    subscription = (
-        AlertRuleQuerySubscription.objects.select_related("query_subscription")
-        .get(alert_rule=alert_rule)
-        .query_subscription
-    )
 
-    alert_rule.update(
-        # Randomize the name here so that we don't get unique constraint issues
-        # while waiting for the deletion to process
-        name=uuid4().get_hex(),
-        status=AlertRuleStatus.PENDING_DELETION.value,
-    )
+    with transaction.atomic():
+        alert_rule.update(
+            # Randomize the name here so that we don't get unique constraint issues
+            # while waiting for the deletion to process
+            name=uuid4().get_hex(),
```

Comment:
Is there an undelete for alert rules, where we'd need to restore the name?

Reply:
Not at the moment. We can build that separately if needed.